lib/Transforms/HandshakeOptimizeBitwidths.cpp

```diff
@@ -657,23 +660,34 @@ struct HandshakeMuxSelect : public OpRewritePattern<handshake::MuxOp> {
     if (optWidth >= selectWidth)
       return failure();
 
-    // Replace the select operand with one with optimize bitwidth
-    Value newSelect =
-        modVal({selectOperand, ExtType::LOGICAL}, optWidth, rewriter);
-    rewriter.replaceUsesWithIf(selectOperand, newSelect, [&](OpOperand &oprd) {
-      return oprd.getOwner() == muxOp;
-    });
+    // Create a new mux whose select operand is optimized
+    SmallVector<Value, 3> newOperands;
+    newOperands.push_back(
+        modVal({selectOperand, ExtType::LOGICAL}, optWidth, rewriter));
```

Comment:
A suggestion that is irrelevant to this PR: it would be nice to call `modVal` something more intuitive as it is kind of the most used utility in this pass, how about `modChannelBitWidth` or `modBitWidth`?